### PR TITLE
Add option to zoom on mouseover on desktop

### PR DESCRIPTION
**Pull request by rerissondaniel**
#### What is the purpose of this pull request?

Allow the user to choose if the zoom will be enabled in on hover or on  click
#### What problem is this solving?

The need for a more dynamic behavior for zoom on desktop.

#### How should this be manually tested?
[workspace on hover](https://zoomonhover--storecomponents.myvtex.com/classic-shoes/p)
[workspace on click](https://zoomonclick--storecomponents.myvtex.com/classic-shoes/p)
[workspace gallery](https://zoomongallery--storecomponents.myvtex.com/classic-shoes/p)
(forced prop value throught code since cms is not persisting changes, but it should work using blocks.json)
#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [X] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/ProductImages/components/Carousel/index.js [hunk 1]
```diff
@@ -288,6 +298,18 @@ class Carousel extends Component {
       }
     )
 
+    const zoomListeners = zoomType === 'in-page' &&
+      desktop &&
+      desktopTrigger === 'on-hover' && {
+        onMouseMove: e => {
+          this.slideZoom.in(e)
+          // I know this is is weird... But it was the only way I've found to
```
Comment (luciannojunior) [inline react/components/ProductImages/components/Carousel/index.js]: Remove the comment or add an FIXME with your name

File: react/components/ProductImages/index.js [hunk 2]
```diff
@@ -120,6 +121,18 @@ ProductImages.getSchema = ({ zoomProps: { zoomType } = {} }) => {
               default: 0.8,
             },
           }),
+          ...(zoomType === 'in-page' && {
+            desktopTrigger: {
```
Comment (luciannojunior) [inline react/components/ProductImages/index.js]: Did you check if this props should have an `isLayout` option?

File: react/components/ProductImages/components/Carousel/index.js (hunk 1 above)
Comment (luciannojunior) [reply react/components/ProductImages/components/Carousel/index.js]: BTW, doesn't look weird

File: react/components/ProductImages/index.js (hunk 2 above)
Comment (rerissondaniel) [reply react/components/ProductImages/index.js]: They should, but this option is currently not working on the cms

**Pull request closed without merging**